### PR TITLE
Fixed the citation in `dominance.py` [Issue #7522]

### LINKED ISSUE #7522: wrong citation for dominance algorithms
### Current Behavior

In https://github.com/networkx/networkx/blob/main/networkx/algorithms/dominance.py the current citation is non-existing:
```
.. [1] K. D. Cooper, T. J. Harvey, and K. Kennedy.
           A simple, fast dominance algorithm.
           Software Practice & Experience, 4:110, 2001.
```

### Expected Behavior

The correct citation is:
```
K.D. Cooper, T.J. Harvey, and K. Kennedy, “A Simple, Fast Dominance Algorithm,” Rice University, Department of Computer Science Technical Report 06-33870, 2006.  [pdf version](http://www.cs.rice.edu/~keith/Publications/TR06-33870-Dom.pdf)
```
Taken from the first authors personal website at http://keith.rice.edu/publications-2/technical-reports/

### PR DESCRIPTION
**Pull request by muddi900**
Fixes #7522

### CODE REVIEW

File: networkx/algorithms/dominance.py [hunk 1]
```diff
@@ -50,9 +50,8 @@ def immediate_dominators(G, start):
 
     References
     ----------
-    .. [1] K. D. Cooper, T. J. Harvey, and K. Kennedy.
-           A simple, fast dominance algorithm.
-           Software Practice & Experience, 4:110, 2001.
+    .. [1] Cooper, Keith D., Harvey, Timothy J. and Kennedy, Ken. "A simple, fast dominance algorithm." (2006).
+https://hdl.handle.net/1911/96345
```
Comment (dschult) [inline networkx/algorithms/dominance.py:54]: I think the linting tool wants this to be wrapped something like this (and for the other function too).
```suggestion
    .. [1] Cooper, Keith D., Harvey, Timothy J. and Kennedy, Ken,
           "A simple, fast dominance algorithm." (2006).
           https://hdl.handle.net/1911/96345
```

Comment (Peiffap) [reply networkx/algorithms/dominance.py:54]: Shouldn't it always be 4-space indents?

```suggestion
    .. [1] Cooper, Keith D., Harvey, Timothy J. and Kennedy, Ken.
        "A simple, fast dominance algorithm." (2006).
        https://hdl.handle.net/1911/96345
```

Comment (dschult) [reply networkx/algorithms/dominance.py:54]: I don't think "always" is a good rule for most things. :)

Sphinx works either way for the html, and lining up the indents looks better when reading the ascii doc_string as part of the file. I would bet it isn't consistent throughout the library and that's ok. But that goals should be:
- readable inline doc_string while reading the code, and 
- nice html output when processed by Sphinx.

**Comment by muddi900** [comment]
I have committed the requested changes.

File: networkx/algorithms/dominance.py [hunk 2]
```diff
@@ -50,9 +50,9 @@ def immediate_dominators(G, start):
 
     References
     ----------
-    .. [1] K. D. Cooper, T. J. Harvey, and K. Kennedy.
-           A simple, fast dominance algorithm.
-           Software Practice & Experience, 4:110, 2001.
+    .. [1] Cooper, Keith D., Harvey, Timothy J. and Kennedy, Ken.
+        "A simple, fast dominance algorithm." (2006).
+        https://hdl.handle.net/1911/96345
```
Comment (Peiffap) [inline networkx/algorithms/dominance.py:55]:
```suggestion
    .. [1] Cooper, Keith D., Harvey, Timothy J. and Kennedy, Ken.
           "A simple, fast dominance algorithm." (2006).
           https://hdl.handle.net/1911/96345
```

File: networkx/algorithms/dominance.py [hunk 3]
```diff
@@ -118,9 +118,8 @@ def dominance_frontiers(G, start):
 
     References
     ----------
-    .. [1] K. D. Cooper, T. J. Harvey, and K. Kennedy.
-           A simple, fast dominance algorithm.
-           Software Practice & Experience, 4:110, 2001.
+    .. [1] Cooper, Keith D., Harvey, Timothy J. and Kennedy, Ken. "A simple, fast dominance algorithm." (2006).
+https://hdl.handle.net/1911/96345
```
Comment (Peiffap) [inline networkx/algorithms/dominance.py:122]:
```suggestion
    .. [1] Cooper, Keith D., Harvey, Timothy J. and Kennedy, Ken.
           "A simple, fast dominance algorithm." (2006).
           https://hdl.handle.net/1911/96345
```

**Review by Peiffap — COMMENTED** [review]
Let's go with Dan's recommendation instead - and apply it to both.

**Comment by muddi900** [comment]
I have applied the changes and flattened the branch

**Review by Schefflera-Arboricola — APPROVED** [review]
Thanks @muddi900 !

Pls consider fixing this citation in the comments of [test_dominance.py](https://github.com/networkx/networkx/blob/main/networkx/algorithms/tests/test_dominance.py) file.

And next time remember to run `pre-commit` after you do `git add .`

**Comment by Peiffap** [comment]
To add on to that, you can run `pre-commit install` which will automatically run the relevant pre-commit hooks whenever you try to `git commit`.

**Comment by Peiffap** [comment]
Could you apply the same indent rules for the citations in the test files? Thanks

**Comment by Peiffap** [comment]
The test citations still have one space too many.

File: networkx/algorithms/tests/test_dominance.py [hunk 4]
```diff
@@ -39,19 +39,21 @@ def test_unreachable(self):
...
     def test_irreducible1(self):
-        # Graph taken from Figure 2 of
-        # K. D. Cooper, T. J. Harvey, and K. Kennedy.
-        # A simple, fast dominance algorithm.
-        # Software Practice & Experience, 4:110, 2001.
+        """
+        .. [1]  Cooper, Keith D., Harvey, Timothy J. and Kennedy, Ken.
+                "A simple, fast dominance algorithm." (2006).
+                https://hdl.handle.net/1911/96345
+        """
```
Comment (rossbar) [inline networkx/algorithms/tests/test_dominance.py:46]: Let's keep the info on figure provenance. No need to worry about formatting here as the test suite is not included in the docs build:

```suggestion
        """
        Graph taken from figure 2 of "A simple, fast dominance algorithm." (2006).
        https://hdl.handle.net/1911/96345
        """
```

File: networkx/algorithms/tests/test_dominance.py [hunk 5]
```diff
@@ -39,19 +39,20 @@ def test_unreachable(self):
...
 
     def test_irreducible1(self):
-        # Graph taken from Figure 2 of
-        # K. D. Cooper, T. J. Harvey, and K. Kennedy.
-        # A simple, fast dominance algorithm.
-        # Software Practice & Experience, 4:110, 2001.
+        """
+        Graph taken from figure 2 of "A simple, fast dominance algorithm." (2006).
+        https://hdl.handle.net/1911/96345
+        """
```
Comment (Peiffap) [inline networkx/algorithms/tests/test_dominance.py:45]: The other test references should be changed similarly.

**Comment by muddi900** [comment]
I have fixed the other citation as well. Would you like me to flatten the commits?

File: networkx/algorithms/tests/test_dominance.py [hunk 6]
```diff
@@ -113,10 +114,11 @@ def test_unreachable(self):
         assert nx.dominance_frontiers(G, n // 2) == {i: set() for i in range(n // 2, n)}
 
     def test_irreducible1(self):
-        # Graph taken from Figure 2 of
-        # K. D. Cooper, T. J. Harvey, and K. Kennedy.
-        # A simple, fast dominance algorithm.
-        # Software Practice & Experience, 4:110, 2001.
+        """
+        .. [1]  Cooper, Keith D., Harvey, Timothy J. and Kennedy, Ken.
```
Comment (Peiffap) [inline networkx/algorithms/tests/test_dominance.py:118]: This one.

File: networkx/algorithms/tests/test_dominance.py [hunk 7]
```diff
@@ -128,10 +130,11 @@ def test_irreducible1(self):
         }
 
     def test_irreducible2(self):
-        # Graph taken from Figure 4 of
-        # K. D. Cooper, T. J. Harvey, and K. Kennedy.
-        # A simple, fast dominance algorithm.
-        # Software Practice & Experience, 4:110, 2001.
+        """
+        .. [1]  Cooper, Keith D., Harvey, Timothy J. and Kennedy, Ken.
```
Comment (Peiffap) [inline networkx/algorithms/tests/test_dominance.py:134]: And this one.

**Review by Peiffap — COMMENTED** [review]
There are still two references that need to be changed in the test file.

I don't think you need to squash the commits yourself; it gets done when merging anyways.

**Review by rossbar — APPROVED** [review]
Thanks @muddi900 and @Peiffap & @Schefflera-Arboricola for reviewing!

**Pull request merged**